### PR TITLE
feat(firestore): Multiple inequalities

### PR DESCRIPTION
**Pull request by bhshkh**
## Description

Created samples for multiple inequalities feature. Currently, it is hardcoded [here](https://firebase.google.com/docs/firestore/query-data/multiple-range-fields#go). 
The sample follows the [style guide](https://googlecloudplatform.github.io/samples-style-guide/) and [contributing guidelines]((https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md))

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
Cloud Firestore API needs to be enabled
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: firestore/query_multiple_inequality.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	"cloud.google.com/go/firestore"
+)
+
+func multipleInequalities(w io.Writer, projectID string) error {
+	ctx := context.Background()
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %v", err)
+	}
```
Comment (subfuzion) [inline firestore/query_multiple_inequality.go:31]: Please add `defer client.Close()`

**Review by subfuzion — CHANGES_REQUESTED** [review]
Thanks @bhshkh, just one requested change.

File: firestore/query_multiple_inequality.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"io"
+
+	"cloud.google.com/go/firestore"
+)
+
+func multipleInequalities(w io.Writer, projectID string) error {
+	ctx := context.Background()
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %v", err)
```
Comment (subfuzion) [inline firestore/query_multiple_inequality.go:30]: Actually, one more change, please: change the verb to "%w" to wrap the error. For samples testing, this helps preserve the error payload for CI logs.

**Review by subfuzion — CHANGES_REQUESTED** [review]

File: firestore/query_multiple_inequality.go (hunk 1 above)
Comment (bhshkh) [reply firestore/query_multiple_inequality.go:33]: Added

File: firestore/query_multiple_inequality.go (hunk 2 above)
Comment (bhshkh) [reply firestore/query_multiple_inequality.go]: Added

**Review by gkevinzheng — APPROVED** [review]

**Review by jiayou97 — APPROVED** [review]
Hi, this is Jia from Firestore team. Thanks for making the change!

**Review by subfuzion — APPROVED** [review]

**Comment by KasiaStrz** [comment]
Hi @bhshkh , could you please add the Datastore version with region tag: datastore_query_filter_compound_multi_ineq? Thanks!

**Pull request merged**